tests/cross-system.bats

```diff
@@ -0,0 +1,105 @@
+
+#! /usr/bin/env bats
+# -*- mode: bats; -*-
+# ============================================================================ #
+#
+# Test if cross system push/pull works.
+#
+#
+# ---------------------------------------------------------------------------- #
+
+load test_support.bash
+
+# bats file_tags=end2end
+
+# ---------------------------------------------------------------------------- #
+
+setup_file() {
+  common_file_setup
+  "$FLOX_BIN" config --set floxhub_url "https://saas-dev-main.floxdev.com/"
+  if [ -z "${FLOXTEST_FLOXHUB_TOKEN:-}" ]; then
+    skip "FLOXTEST_FLOXHUB_TOKEN is not set"
+  fi
+  export FLOX_FLOXHUB_TOKEN="$FLOXTEST_FLOXHUB_TOKEN"
+  export OWNER="floxtest"
+}
+
+teardown_file() {
+  "$FLOX_BIN" config --delete floxhub_url
+  unset FLOX_FLOXHUB_TOKEN
+}
+
+# ---------------------------------------------------------------------------- #
+
+# Helpers for project based tests.
+
+project_setup() {
+  export PROJECT_DIR="${BATS_TEST_TMPDIR?}/project-${BATS_TEST_NUMBER?}"
+  export PROJECT_NAME="${PROJECT_DIR##*/}"
+  rm -rf "$PROJECT_DIR"
+  mkdir -p "$PROJECT_DIR"
+  pushd "$PROJECT_DIR" > /dev/null || return
+}
+
+project_teardown() {
+  popd > /dev/null || return
+  rm -rf "${PROJECT_DIR?}"
+  unset PROJECT_DIR
+  unset PROJECT_NAME
+}
+
+# ---------------------------------------------------------------------------- #
+
+setup() {
+  common_test_setup
+  project_setup
+}
+teardown() {
+  project_teardown
+  common_test_teardown
+}
+
+# ---------------------------------------------------------------------------- #
+
+@test "flox push succeeds" {
+  name="created-on-$NIX_SYSTEM"
+
+  "$FLOX_BIN" init -n "$name"
+  "$FLOX_BIN" install hello
+  run "$FLOX_BIN" push --owner "$OWNER" --force
+  assert_success
+  assert_output --partial "pushed to floxhub (forced)"
+}
+
+# This should pull the environment created by the previous run on a different
+# system of the flox push test above.
+# Because we don't check for anything other than hello being installed,
+# hopefully race conditions won't be an issue.
+@test "can flox pull and activate an environment created on another system" {
+  local pull_system
+  case "$NIX_SYSTEM" in
+    x86_64-linux)
+      pull_system="aarch64-darwin"
+      ;;
+    aarch64-linux)
+      pull_system="x86_64-darwin"
+      ;;
+    x86_64-darwin)
+      pull_system="aarch64-linux"
+      ;;
+    aarch64-darwin)
+      pull_system="x86_64-linux"
+      ;;
+    *)
+      echo "unsupported system: $NIX_SYSTEM"
+      return 1
+      ;;
+  esac
+
+  name="created-on-$pull_system"
+
+  "$FLOX_BIN" pull "$OWNER/$name" --amend-system
```

Comment:
I think that's covered in cli/tests. This test is a bit more expensive as it's talking to saas-dev-main.floxdev.com, so I'm thinking it might be best to keep the cases here more minimal?